src/libtiled/tilelayer.h

```diff
@@ -513,6 +515,7 @@ class TILEDSHARED_EXPORT TileLayer : public Layer
     int mHeight;
     Cell mEmptyCell;
     QHash<QPoint, Chunk> mChunks;
+    QRegion mBounds;
```

Comment:
I previously tried with `QRect`, but we cannot do arithmetic operations on `QRect`. Although we can convert `QRect` to `QRegion`, then add a chunk and then use `QRegion::boundingRect` but I guess that would have the same complexity.